File: Preparation Kit/Week 1/5_Camel_Case.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

void splitCamelCase(char* input, char type) {
    if (type == 'M') {
        input[strlen(input) - 2] = '\0';  // Remove "()"
    }
    if (type == 'C') {
        printf("%c", tolower(input[0]));
    } else {
        printf("%c", input[0]);
    }

    for (int i = 1; input[i] != '\0'; i++) {
        if (isupper(input[i])) {
            printf(" ");
            printf("%c", tolower(input[i]));
        } else {
            printf("%c", input[i]);
        }
    }
    printf("\n");
}

void combineCamelCase(char* input, char type) {
    if (type == 'C') {
        printf("%c", toupper(input[0]));
    } else {
        printf("%c", input[0]);
    }
    int flag = 0;
    
    for (int i = 1; input[i] != '\0'; i++) {
        if (input[i] == ' ') {
            flag = 1;
        } else {
            if (flag == 1) {
                printf("%c", toupper(input[i]));
                flag = 0;
            }
            else {
                printf("%c", input[i]);
            }
        }
    }
    if (type == 'M') printf("()");
    printf("\n");
}

int main() {

    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    
    char line[1024];

    while (fgets(line, sizeof(line), stdin)) {
        line[strcspn(line, "\r\n")] = '\0';

        char operation, type;
        char input[1024];

        if (sscanf(line, "%c;%c;%[^\n]", &operation, &type, input) == 3) {

            if (operation == 'S') {
                splitCamelCase(input, type);
            } else if (operation == 'C') {
                combineCamelCase(input, type);
            }
        }
    }

    return 0;
}

File: Preparation Kit/Week 1/5_Camel_Case.py
# Enter your code here. Read input from STDIN. Print output to STDOUT

import re
import sys

def split_camel_case(input_str, type_):
    if type_ == "M":
        # Remove "()" for methods
        input_str = input_str[:-2]  
    words = re.findall(r'[A-Za-z][^A-Z]*', input_str)
    return ' '.join(word.lower() for word in words)

def combine_camel_case(input_str, type_):
    words = input_str.split()
    if type_ == "C":
        # Class: Every word capitalized
        return ''.join(word.capitalize() for word in words)
    elif type_ == "V":
        # Variable: First word lowercase, rest capitalized
        return words[0].lower() + ''.join(word.capitalize() for word in words[1:])
    elif type_ == "M":
        # Method: First word lowercase, rest capitalized, ends with "()"
        return words[0].lower() + ''.join(word.capitalize() for word in words[1:]) + "()"

# Read all input and process each line
for line in sys.stdin:
    operation, type_, input_str = line.strip().split(';')
    if operation == "S":
        sys.stdout.write(split_camel_case(input_str, type_) + "\n")
    elif operation == "C":
        sys.stdout.write(combine_camel_case(input_str, type_) + "\n")
